### PR TITLE
Remove `contain` layout and `canvasRatio` prop

### PR DESCRIPTION
**Pull request by axelboc**
Part of #1275. Simplifies a lot of things! As discussed someone who wants to get the axes flush against an orthonormal heatmap can always compute the right width/height to give to the canvas' parent element with hard-coded axis offset values.

### CODE REVIEW

File: packages/app/src/vis-packs/core/heatmap/models.ts [hunk 1]
```diff
@@ -1,5 +1,3 @@
 import type { INTERPOLATORS } from '@h5web/lib';
 
 export type ColorMap = keyof typeof INTERPOLATORS;
-
-export type Layout = 'contain' | 'cover' | 'fill';
```
Comment (axelboc) [inline packages/app/src/vis-packs/core/heatmap/models.ts:5]: This was already declared in the lib.

File: packages/lib/src/vis/shared/VisCanvas.tsx [hunk 2]
```diff
@@ -60,50 +52,42 @@ function VisCanvas(props: PropsWithChildren<Props>) {
         paddingRight: axisOffsets.right,
       }}
     >
-      {(!shouldMeasure || canvasSize) && (
-        <div
-          className={styles.canvasWrapper}
-          style={shouldMeasure ? canvasSize : undefined}
+      <div className={styles.canvasWrapper}>
```
Comment (loichuder) [inline packages/lib/src/vis/shared/VisCanvas.tsx:55]: Perhaps this `div` is no longer needed ?

**Review by loichuder — APPROVED** [review]
Sweet!

**Comment by loichuder** [comment]
Just wait before we do a fix release before merging this breaking change.

File: packages/lib/src/vis/shared/VisCanvas.tsx (hunk 2 above)
Comment (axelboc) [reply packages/lib/src/vis/shared/VisCanvas.tsx]: I was able to make a bunch of improvements. I'll comment.

File: apps/storybook/src/VisCanvas.stories.tsx [hunk 3]
```diff
@@ -34,9 +34,9 @@ LogScales.args = {
 
 export const AspectRatio = Template.bind({});
 AspectRatio.args = {
-  abscissaConfig: { visDomain: [0, 16], showGrid: true, isIndexAxis: true },
+  abscissaConfig: { visDomain: [0, 20], showGrid: true, isIndexAxis: true },
   ordinateConfig: { visDomain: [0, 10], showGrid: true, isIndexAxis: true },
-  canvasRatio: 16 / 10,
+  visRatio: 20 / 10,
```
Comment (axelboc) [inline apps/storybook/src/VisCanvas.stories.tsx:39]: This was still testing `canvasRatio`, so I've changed it to `visRatio`.

File: packages/lib/src/vis/shared/AxisSystem.module.css [hunk 4]
```diff
@@ -1,5 +1,9 @@
 .axisSystem {
   position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
```
Comment (axelboc) [inline packages/lib/src/vis/shared/AxisSystem.module.css:6]: See `AxisSystem` below.

File: packages/lib/src/vis/shared/AxisSystem.tsx [hunk 5]
```diff
@@ -29,15 +26,10 @@ function AxisSystem(props: Props) {
 
   return (
     // Append to `canvasWrapper` instead of default container `r3fRoot`, which hides overflow
-    <Html container={gl.domElement.parentElement?.parentElement || undefined}>
+    <Html overflowCanvas>
```
Comment (axelboc) [inline packages/lib/src/vis/shared/AxisSystem.tsx:29]: We weren't taking advantage of the prop we added specifically for this.

File: packages/lib/src/vis/shared/VisCanvas.tsx [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
-import { getSizeToFit, getAxisOffsets } from '../utils';
+import { getAxisOffsets } from '../utils';
 import AxisSystem from './AxisSystem';
 import AxisSystemProvider from './AxisSystemProvider';
 import RatioEnforcer from './RatioEnforcer';
 import ThresholdAdjuster from './ThresholdAdjuster';
 import ViewportCenterer from './ViewportCenterer';
 import styles from './VisCanvas.module.css';
 
+const NO_OFFSETS = { left: 0, right: 0, top: 0, bottom: 0 };
```
Comment (loichuder) [inline packages/lib/src/vis/shared/VisCanvas.tsx:15]: Good call :+1:

**Review by loichuder — APPROVED** [review]

File: packages/lib/src/vis/shared/AxisSystem.tsx [hunk 7]
```diff
@@ -29,15 +26,10 @@ function AxisSystem(props: Props) {
...
       <div
         className={styles.axisSystem}
         style={{
-          // Take over space reserved for axis by VisCanvas
-          top: -axisOffsets.top,
-          left: -axisOffsets.left,
-          width: width + axisOffsets.left + axisOffsets.right,
-          height: height + axisOffsets.bottom + axisOffsets.top,
```
[the remark this answers is not in the document]
Comment (axelboc) [reply packages/lib/src/vis/shared/AxisSystem.tsx:40]: Indeed, indeed 😁

**Pull request merged**